### PR TITLE
Fantom connectors deployment

### PR DESCRIPTION
**Pull request by Sowmayjain**
- [x] Review and approve connector
- [x] Deploy the contracts and comment the addresses
- [x] Created resolver - (add PR link here) [PR](https://github.com/Instadapp/dsa-resolvers/pull/57)
- [x] Add to the dsa-connect - [PR](https://github.com/Instadapp/dsa-connect/pull/129)
- [ ] Add to the docs - link the PR

### CODE REVIEW

**Comment by bhavik-m** [comment]
Deployed Connector Address : 

-  `AUTHORITY-A`: [0x6CE3e607C808b4f4C26B7F6aDAeB619e49CAbb25](https://ftmscan.com/address/0x6CE3e607C808b4f4C26B7F6aDAeB619e49CAbb25), [tx](https://ftmscan.com/tx/0x2082988d6d993ac8e8495a91d44eae5beb784f326f8072b8c3f8e371a0ffd76b)
-   `BASIC-A`: [0x9926955e0dd681dc303370c52f4ad0a4dd061687](https://ftmscan.com/address/0x9926955e0dd681dc303370c52f4ad0a4dd061687),  [tx](https://ftmscan.com/tx/0x2082988d6d993ac8e8495a91d44eae5beb784f326f8072b8c3f8e371a0ffd76b)

- `BASIC-B` :  [0x8dA60dee0815a08d16C066b07814b10722fA9306](https://ftmscan.com/address/0x8dA60dee0815a08d16C066b07814b10722fA9306), [tx](https://ftmscan.com/tx/0x677e7bfa9a25a58b9a4b916dde567a0eeff3d9ed9dcc31e11898f303186dc90d#eventlog)
- `BASIC-C`  :   [0x2EadEecf1aB9283a9F76D219CF17eeD4932B8811](https://ftmscan.com/address/0x2EadEecf1aB9283a9F76D219CF17eeD4932B8811), [tx](https://ftmscan.com/tx/0x677e7bfa9a25a58b9a4b916dde567a0eeff3d9ed9dcc31e11898f303186dc90d#eventlog)
- `WFTM-A` : [0x1310f478A37bBdccf47E8e10962d3bB888FEc62C](https://ftmscan.com/address/0x1310f478A37bBdccf47E8e10962d3bB888FEc62C#code), [tx](https://ftmscan.com/tx/0x677e7bfa9a25a58b9a4b916dde567a0eeff3d9ed9dcc31e11898f303186dc90d#eventlog)

File: .gitignore [hunk 1]
```diff
@@ -63,4 +63,8 @@ build/contracts
 # buidler
 artifacts
 cache
-typechain
\ No newline at end of file
+<<<<<<< HEAD
+typechain
+=======
+typechain
+>>>>>>> 64d52604ec3f9c9172250d8ab60275599074af87
```
Comment (thrilok209) [inline .gitignore]: @bhavik-m What is this?

File: contracts/fantom/connectors/wftm/main.sol [hunk 2]
```diff
@@ -0,0 +1,65 @@
+pragma solidity ^0.7.0;
+
+/**
+ * @title WETH.
+ * @dev Wrap and Unwrap WETH.
```
Comment (thrilok209) [inline contracts/fantom/connectors/wftm/main.sol]: @bhavik-m WFTM

File: contracts/fantom/connectors/wftm/main.sol [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+import { DSMath } from "../../common/math.sol";
+import { Basic } from "../../common/basic.sol";
+import { Events } from "./events.sol";
+import { Helpers } from "./helpers.sol";
+
+abstract contract Resolver is Events, DSMath, Basic, Helpers {
+
+    /**
+     * @dev Deposit ETH into WETH.
+     * @notice Wrap ETH into WETH
```
Comment (thrilok209) [inline contracts/fantom/connectors/wftm/main.sol]: @bhavik-m Here too

File: contracts/fantom/connectors/wftm/main.sol [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+        setUint(setId, _amt);
+
+        _eventName = "LogWithdraw(uint256,uint256,uint256)";
+        _eventParam = abi.encode(_amt, getId, setId);
+    }
+}
+
+contract ConnectV2WETHArbitrum is Resolver {
+    string constant public name = "WETH-v1.0";
```
Comment (thrilok209) [inline contracts/fantom/connectors/wftm/main.sol]: @bhavik-m

File: contracts/fantom/connectors/basic/main.sol [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+import { Events } from "./events.sol";
+
+abstract contract BasicResolver is Events, DSMath, Basic {
+    using SafeERC20 for IERC20;
+
+    /**
+     * @dev Deposit Assets To Smart Account.
+     * @notice Deposit a token to DSA
+     * @param token The address of the token to deposit. (For MATIC: 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE)
+     * @param amt The amount of tokens to deposit. (For max: `uint256(-1)` (Not valid for MATIC))
```
Comment (thrilok209) [inline contracts/fantom/connectors/basic/main.sol:23]: @bhavik-m

File: contracts/fantom/connectors/basic/main.sol [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+        setUint(setId, _amt);
+
+        _eventName = "LogDeposit(address,uint256,uint256,uint256)";
+        _eventParam = abi.encode(token, _amt, getId, setId);
+    }
+
+    /**
+     * @dev Withdraw Assets from Smart  Account
+     * @notice Withdraw a token from DSA
+     * @param token The address of the token to withdraw. (For MATIC: 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE)
```
Comment (thrilok209) [inline contracts/fantom/connectors/basic/main.sol:51]: @bhavik-m

**Review by pradyuman-verma — DISMISSED** [review]

**Review by pradyuman-verma — APPROVED** [review]

**Pull request merged**